src/endpoint/s3/ops/s3_put_bucket_lifecycle.js

```diff
@@ -171,10 +173,14 @@ async function put_bucket_lifecycle(req) {
         return current_rule;
     });
 
-    await req.object_sdk.set_bucket_lifecycle_configuration_rules({
-        name: req.params.bucket,
-        rules: lifecycle_rules
-    });
+    try {
+        await req.object_sdk.set_bucket_lifecycle_configuration_rules({
+            name: req.params.bucket,
+            rules: lifecycle_rules
+        });
+    } catch (error) {
+        s3_utils.invalid_schema_to_aws_error(error);
```

Comment:
I didn't understand what is the example of getting this error in AWS? (as we are trying to be compatible to them), If you can reference the screenshot or attach it here.

Reply:
Its an extra check where we get `invalid_schema_params` or `invalid_schema` error for any edge case, we are returning an aws error(MalformedPolicy). Not sure, if we need it or not but it was there already in `put_bucket_policy` so I have added it here in `put_bucket_lifecycle_configuration` as well.

Reply:
I'm not sure you need to add it as you saw it in `put_bucket_policy` and not in `put_bucket_lifecycle_configuration `.

Reply:
I will remove it from here then and will add it if needed in future. Thanks